contents/data/databases.md

```diff
@@ -0,0 +1,115 @@
+
+However, having such a rigid structure would mean that there's a limitation on how and what kind of data can be stored in the database.
+
+For the purposes of introduction, we would mainly cover on aspects that are used in the relational and/or document-oriented database models.
+
+## Relation
+
+The most popular database model, relational model, makes use of relations. This model assumes that the data to be stored follow a certain 'pattern'. For instance, a database that stores products sold by a shop would contain data such as: name, description, price and current stock levels. A visualisation of the database and data can be seen below:
+
+| Name  | Description                  | Price | Current Stock |
+|:-----:|:----------------------------:|:-----:|:-------------:|
+| Fruit | A fruit.                     | 1.00  |      100      |
+| Bread | Sliced for your convenience. | 1.40  |       50      |
+| Water | Essential for life.          | 0.50  |     1000      |
+
+From this visualisation, we can define the different parts of the relation:
+
+- The whole table is known as a **relation**.
+- Each data row (i.e. excluding the header row) is known as a **tuple**, for instance: (Fruit, A fruit., 1.00, 100).
+- Each column is an **attribute** and each attribute has a **domain** or **data type**. The *Current Stock* column of the table is an attribute with a data type of integer (as product stock is logically represented using an integer).
+- Each element in a tuple is called an **attribute value**
+
+[Click here](https://en.wikipedia.org/wiki/Relation_(database)) for a more in-depth and formal definition of relations.
+
+## Transactions
+
+Databases are useless if the data cannot be used in a meaningful manner. However, uncontrolled access to the database would not be ideal, as it might cause problems, especially when other actions may depend on the previous action's result.
+
+Take for instance, a program that transfers money from one bank account to another. The actions that the program would need to do involves:
+
+1. Checking if the sender has enough money
+2. Deducting the amount of money to be sent
+3. Adding that amount to the receiver's account
+
+A problem arises if another action takes place in between any of the steps (e.g. another transfer from the same sender), or if any of the steps fails (e.g. due to a program crash). As a result, it may cause the data to be manipulated in an undesired manner (e.g. money not credited to receiver).
+
+Thus, **transactions** allow us to guard against these problems. A **transaction** symbolises a logical unit of work, which consists of multiple database actions, performed on a set of databases. [[Source]](https://en.wikipedia.org/wiki/Database_transaction) Properties of a database transaction ensures that these actions are done in a predictable and reliable manner.
```

Comment:
could talk about how transaction being a logical unit of work, ensures that the steps in the transaction either happen in the sequence defined or it didn't happen at all. (: